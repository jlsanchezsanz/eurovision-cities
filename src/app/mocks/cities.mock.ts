import { CitiesResponse } from '../models';

export const citiesResponseMock: CitiesResponse = {
  content: [
    { id: 317, name: 'Ahvaz' },
    { id: 87, name: 'Al-Jizah' },
    { id: 147, name: 'Al-Khartum Bahri' },
    { id: 227, name: 'Al-Madinah' },
    { id: 103, name: 'Aleppo' },
    { id: 47, name: 'Alexandria' },
    { id: 176, name: 'Algiers' },
    { id: 307, name: 'Allahabad' },
    { id: 302, name: 'Almaty' },
    { id: 260, name: 'Amman' },
  ],
  pageable: {
    sort: { sorted: false, unsorted: true, empty: true },
    pageSize: 10,
    pageNumber: 1,
    offset: 10,
    unpaged: false,
    paged: true,
  },
  totalPages: 34,
  totalElements: 332,
  last: false,
  first: false,
  sort: { sorted: false, unsorted: true, empty: true },
  numberOfElements: 10,
  size: 10,
  number: 1,
  empty: false,
};
