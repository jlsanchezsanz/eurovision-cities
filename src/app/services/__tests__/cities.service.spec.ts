import { CitiesService } from '../cities.service';
import { citiesResponseMock } from '../../mocks';

describe('CitiesService', () => {
  let citiesService: CitiesService;
  const httpMock = {
    get: jest.fn(),
  } as any;

  beforeEach(() => {
    citiesService = new CitiesService(httpMock);
  });

  it('should fetch pets data', () => {
    httpMock.get.mockReturnValueOnce(citiesResponseMock);
    expect(httpMock.get).not.toHaveBeenCalled();
    const result = citiesService.getCities(1, 10);
    expect(httpMock.get).toHaveBeenCalledWith(
      'http://localhost:1111/cities/queryByPage?page=1&size=10'
    );
    expect(result).toMatchSnapshot();
  });
});
