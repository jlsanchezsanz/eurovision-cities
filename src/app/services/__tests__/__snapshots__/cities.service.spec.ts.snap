// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesService should fetch pets data 1`] = `
Object {
  "content": Array [
    Object {
      "id": 317,
      "name": "Ahvaz",
    },
    Object {
      "id": 87,
      "name": "Al-Jizah",
    },
    Object {
      "id": 147,
      "name": "Al-Khartum Bahri",
    },
    Object {
      "id": 227,
      "name": "Al-Madinah",
    },
    Object {
      "id": 103,
      "name": "Aleppo",
    },
    Object {
      "id": 47,
      "name": "Alexandria",
    },
    Object {
      "id": 176,
      "name": "Algiers",
    },
    Object {
      "id": 307,
      "name": "Allahabad",
    },
    Object {
      "id": 302,
      "name": "Almaty",
    },
    Object {
      "id": 260,
      "name": "Amman",
    },
  ],
  "empty": false,
  "first": false,
  "last": false,
  "number": 1,
  "numberOfElements": 10,
  "pageable": Object {
    "offset": 10,
    "pageNumber": 1,
    "pageSize": 10,
    "paged": true,
    "sort": Object {
      "empty": true,
      "sorted": false,
      "unsorted": true,
    },
    "unpaged": false,
  },
  "size": 10,
  "sort": Object {
    "empty": true,
    "sorted": false,
    "unsorted": true,
  },
  "totalElements": 332,
  "totalPages": 34,
}
`;
